File: test-junit5/src/test/java/io/micronaut/test/junit5/DbProperties.java

package io.micronaut.test.junit5;

import io.micronaut.context.annotation.Property;

import java.lang.annotation.Documented;
import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

@Retention(RetentionPolicy.RUNTIME)
@Target({ElementType.METHOD, ElementType.ANNOTATION_TYPE, ElementType.TYPE})
@Documented

@Property(name = "datasources.default.name", value = "testdb")
@Property(name = "jpa.default.properties.hibernate.hbm2ddl.auto", value = "update")
@Property(name = "jpa.default.compileTimeHibernateProxies", value = "true")
public @interface DbProperties {
}

File: test-kotest/src/test/kotlin/io/micronaut/test/kotest/DbProperties.kt

package io.micronaut.test.kotest

import io.micronaut.context.annotation.Property
import io.micronaut.context.annotation.PropertySource

@Retention(AnnotationRetention.RUNTIME)
@Target(AnnotationTarget.FUNCTION, AnnotationTarget.PROPERTY_GETTER, AnnotationTarget.PROPERTY_SETTER, AnnotationTarget.ANNOTATION_CLASS, AnnotationTarget.CLASS, AnnotationTarget.FILE)
@MustBeDocumented
@PropertySource(
        Property(name = "datasources.default.name", value = "testdb"),
        Property(name = "jpa.default.properties.hibernate.hbm2ddl.auto", value = "update")
)
annotation class DbProperties
